### ISSUE TITLE
[acf field="..."] not working in Gutenberg "Query Loop" block

### ISSUE THREAD

**Issue by scrobbleme** [AdvancedCustomFields/acf#645]
As the title says.

It is not possible to output a custom field using the ACF shortcode within the [Query Loop](https://wordpress.org/support/article/query-loop-block/) block.

It just outputs "nothing".

**Comment by mattgrshaw** [comment]
@scrobbleme Thanks for the report!

Just to clarify, are you expecting that the shortcode would output the field values for the individual posts in the query loop, rather than the field value for the post/page that contains the query loop?

In my testing, the shortcode does output the value (if present) for the post/page that contains the query loop, but not the values for any posts/pages inside the query loop. I think this is worth looking into further,  as it definitely doesn't feel intuitive and may not be viable for most folks.

In the meantime, you may be able to achieve something similar with some custom ACF blocks. Liam recently wrote an article on how to do just that here:
https://www.advancedcustomfields.com/blog/build-business-listing-acf-blocks-query-loop-block-theme/

**Comment by scrobbleme** [comment]
@scrobbleme  The first. 
Output the value of the **individual post from the loop.**  (I think this is what most people expect, so should be default)

Output of the containing post might be an option too, maybe through a switch, like `[acf post_id="@outer"]`

There also could be something like `[acf post_id="@loop"]`

**Comment by fisher2470** [comment]
I agree, this should be the default behavior of the short code when used in the query loop block

**Comment by fisher2470** [comment]
For anyone else having issues displaying ACF fields in the query loop block. You can create a custom excerpt using the 'get_the_excerpt' hook. You will have access to the post ID and will be able to isolate the custom excerpt to only the page you'd like this functionality for. Then you can just drop the Post Excerpt block into the query loop.

**Comment by scrobbleme** [comment]
> I think this plugin might work for you: https://wordpress.org/plugins/display-a-meta-field-as-block/
> 
> Or, you could try my general shortcode workaround plugin: https://github.com/KeanuTang/query-loop-shortcode

+ 1 for [Meta Field Block](https://wordpress.org/plugins/display-a-meta-field-as-block/). This plugin is very good. I'm using this for a lot ... I.e., just using a "special" meta key and alter the output through a simple filter.
